tests/test_cases/forkexecuid.c

```diff
@@ -0,0 +1,28 @@
+#include <sys/types.h>
+#include <unistd.h>
+#include <stdio.h>
+
+#include <sys/types.h>
+#include <sys/wait.h>
+
+int main(void)
+{
+  pid_t pid;
+  	puts("printing getgid:");
+	printf("%d\n", getgid());
+	puts("printing getuid:");
+	printf("%d\n", getuid());
+	puts("printing getegid:");
+	printf("%d\n", getegid());
+	puts("printing geteuid:");
+	printf("%d\n", geteuid());
+
+  if ((pid = fork()) == -1) {
+    perror("fork error");
+  }
+  else if (pid == 0) {
+    char* arr[] = {"getuid", NULL};
+    execv("automated_tests/getuid", arr);
+  }
```

Comment:
This syntax here is kind of confusing and not very clean (particularly within the first if condition). I think it would be cleaner to either assign pid before or to have a switch case with cases 0, -1, default

Reply:
Should be fixed, moved pid = fork() to another line.